platforms/core-configuration/model-core/src/main/java/org/gradle/internal/reflect/validation/DefaultTypeAwareProblemBuilder.java

```diff
@@ -123,6 +129,24 @@ public static String introductionFor(Optional<TypeValidationData> additionalData
             builder.append(property)
                 .append("' ");
         }
+
+        Object method = additionalData.map(TypeValidationData::getMethodName).orElse(null);
+        if (method != null) {
+            if (typeRelevant) {
+                builder.append("method '");
+            } else {
+                if (pluginId.isPresent()) {
+                    builder.append("In plugin '")
+                        .append(pluginId.get())
+                        .append("' method '");
+                } else {
+                    builder.append("Method '");
+                }
+            }
+            builder.append(method)
+                .append("' ");
+        }
```

Comment:
Function extracted.

> Is the lack of parentProperty for methods an oversight?

No.  Properties are hierarchical because of the `@Nested` annotation, so a property could be buried deep within a nested set of properties and you can walk the hierarchy starting from the root property.  We don't have this hierarchy with the function annotations right now, so you can really just query/validate the functions present on a given type, and there's no concept of a "parent".